### PR TITLE
Special greeting if player is sick

### PR DESCRIPTION
**Pull request by kaido1224**
Add special greeting if the player told Monika they were sick

### CODE REVIEW

**Comment by ashameimaru** [comment]
SINCE I'M RESPONDING TO THIS while seriously sick with the flu, I might as well go ahead and respond to this one since this is quite pertinent, lol.

For the most part, I am seeing no issues with the dialogue and would be something I would expect Monika to be telling me right now. 

I'm going to comment on different parts for any syntax or other issues.

However, is there an original trigger that tells Monika that you're sick? If so, where is it?

File: Monika After Story/game/script-greetings.rpy [hunk 1]
```diff
@@ -679,3 +679,42 @@ label greeting_japan:
...
+    m 3g "Getting plenty of rest is the best remedy for getting over a sickness quickly."
+    m 2q "I wouldn't forgive myself if your sickness got any worse because of me."
+    m 1p "Now please, [player], put my mind at ease and go get some rest."
+    m 1e "Will you do that for me?"
+    menu:
+        "Yes":
+            jump greeting_stillsickrest
+        "No":
+            jump greeting_stillsicknorest
+
```
Comment (ashameimaru) [inline Monika After Story/game/script-greetings.rpy]: Return needed here.

File: Monika After Story/game/script-ch30.rpy [hunk 2]
```diff
@@ -520,26 +520,32 @@ label ch30_autoload:
...
-    if persistent.closed_self:
-        #pick a random greeting
-        if is_monika_in_room:
-            if persistent.current_monikatopic != "i_greeting_monikaroom":
-                $ pushEvent("i_greeting_monikaroom")
-        else:
-            $pushEvent(renpy.random.choice(greetings_list))
+    #if the player told Monika that they are sick, jump to this greeting.
+    if persistent.sick:
+        $ pushEvent("greeting_sick")
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-ch30.rpy]: You're not supposed to get a greeting if you didn't say good bye to monika, so this should be in the `closed_self` flow, specifically as an elif for the `is_monika_in_room`.

File: Monika After Story/game/script-ch30.rpy [hunk 3]
```diff
@@ -520,26 +520,32 @@ label ch30_autoload:
...
+                    $ pushEvent("i_greeting_monikaroom")
+            else:
+                $pushEvent(renpy.random.choice(greetings_list))
 
-    if not is_monika_in_room:
-        $ set_keymaps()
+        if not persistent.tried_skip:
+            $ config.allow_skipping = True
+        else:
+            $ config.allow_skipping = False
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-ch30.rpy]: Lines 537 to 540 need to run regardless of greeting/no greeting, bring these down an indentation level.

File: Monika After Story/game/script-ch30.rpy [hunk 4]
```diff
@@ -520,26 +520,32 @@ label ch30_autoload:
...
-    $persistent.closed_self = False
-    $startup_check = False
-    jump ch30_loop
+        if not is_monika_in_room:
+            $ set_keymaps()
+
+        $persistent.closed_self = False
+    
+        $startup_check = False
+        jump ch30_loop
```
Comment (ThePotatoGuy) [inline Monika After Story/game/script-ch30.rpy]: Lines 542-548 also need to run regardless of greeting/no greeting, bring theses down an indentation level.

**Comment by kaido1224** [comment]
@ashameimaru it was already merged. This is the other half. 

@ThePotatoGuy  I made the changes you requested but now she doesn't say anything when you open the game after she force closes it.

**Pull request merged**